src/components/FeatureViews/Accounts/ComponentViewer.js

```diff
@@ -25,56 +25,67 @@ const ComponentViewer = ({inView, user}) => {
 			{
 				component: <InstructorSchedule instructorID={user.user_id} key={0}/>,
 				access_permissions: [USER_TYPES.receptionist, USER_TYPES.admin],
+				id: 0,
 			},
 			{
 				component: <InstructorCourses instructorID={user.user_id} key={1}/>,
 				access_permissions: [USER_TYPES.receptionist, USER_TYPES.admin],
+				id: 1,
 			},
 			{
 				component: <Bio background={user.background} key={2}/>,
 				access_permissions: [USER_TYPES.receptionist, USER_TYPES.admin, USER_TYPES.instructor],
+				id: 2,
 			},
 			{
 				component: <StudentCourseViewer current key={3} studentID={user.user_id}/>,
 				access_permissions: [USER_TYPES.receptionist, USER_TYPES.admin],
+				id: 3,
 			},
 			{
 				component: <StudentCourseViewer current={false} key={4} studentID={user.user_id}/>,
 				access_permissions: [USER_TYPES.receptionist, USER_TYPES.admin],
+				id: 4,
 			},
 			{
 				component: <PaymentHistory key={5} user_id={user.user_id}/>,
 				access_permissions: [USER_TYPES.receptionist, USER_TYPES.admin],
+				id: 5,
 			},
 			{
 				component: <ParentContact key={6} parent_id={user.parent_id}/>,
-				access_permissions: [USER_TYPES.receptionist, USER_TYPES.admin, USER_TYPES.student, USER_TYPES.parent],
+				access_permissions: [USER_TYPES.receptionist, USER_TYPES.admin, USER_TYPES.student,],
+				id: 6,
 			},
 			{
 				component: <Notes key={7} ownerID={user.user_id} ownerType="account"/>,
 				access_permissions: [USER_TYPES.receptionist, USER_TYPES.admin, USER_TYPES.student, USER_TYPES.parent, USER_TYPES.instructor],
+				id: 7,
 			},
 			{
 				component: <StudentInfo key={8} user={user}/>,
-				access_permissions: [USER_TYPES.receptionist, USER_TYPES.admin],
+				access_permissions: [USER_TYPES.receptionist, USER_TYPES.admin, USER_TYPES.parent],
+				id: 8,
 			},
 			{
 				component: <PayCourses key={9} user={user}/>,
 				access_permissions: [USER_TYPES.receptionist, USER_TYPES.admin],
+				id: 9,
 			},
 			{
 				component: <UserAccessControl key={10} user={user}>
 					<NotificationSettings user={user}/>
 				</UserAccessControl>,
 				access_permissions: [USER_TYPES.student, USER_TYPES.parent, USER_TYPES.instructor],
+				id: 10,
 			}
 		].filter((tab) => (tab.access_permissions.includes(AuthUser.accountType))),
 		[user]
 	);
 
 	return (
 		<Grid className="profile-component-container">
-			{componentsArray[inView].component}
+			{componentsArray.find(({id}) => id == inView).component}
```

Comment:
Is there a reason for this change?

Reply:
We're filtering down the componentsArray based on which has access to which tabs/component. When we filter it down, the ordering of the components don't match the tab ordering from UserProfile.js (e.g. if Notes has index 5 in UserProfile.js, it may not have index 5 in ComponentViewer when we filter down the array of components there). This way we can specify exactly which tab we're looking for